Read-only mode

awead opened the pull request:
When setting the appropriate environment variable, Scholarsphere will go into "read-only" mode, during which only Solr and database reading operations will work.  Logins are prohibited as well as any other database writing operations.

This is a UI-level change only. Changes via the console are still permissible.

A flash message informs the user that Scholarsphere's content cannot be changed during this time.

Fixes #848 

![Screen Shot 2021-02-05 at 2 37 29 PM](https://user-images.githubusercontent.com/312085/107230054-905acc00-69ec-11eb-83e8-cb32066a2375.png)

awead commented:
@cam156 This fixes the capturing of the stats and the display, but the data gathering has some problems.
![stats](https://cloud.githubusercontent.com/assets/312085/26210627/3b78a546-3bbe-11e7-9147-d1b19355d78b.png)
I've note one issue here #856 but I'm sure there are other issues that we should capture in different tickets.

cam156 commented on app/views/curation_concerns/file_sets/_actions.html.erb:
```diff
@@ -0,0 +1,42 @@
+<div class="btn-group">
```
@awead can you add a comment here indicating that you are overriding the file to allow for download tracking?

awead replied:
@cam156 it's on line 31, just above where the override is occurring.

cam156 replied:
@awead It is easier to find it when it is at the top of the file.  The other files had the change at the top.  Why in this one did you only add it on the line?

awead replied:
@cam156 for the other files, it made more sense to put it near the `if` block, which coincided with the top of the file. In the case of this file, there are many `if can?` blocks and it's not immediately evident that we added an additional one. I wanted the comment to go closest the one that we were adding so that it was more evident what changes were being made.

cam156 replied:
@awead All I am asking for is an additional comment at the top of the file  to make it obvious the only reason this file was overridden was for download tracking.  Are you not willing to put that comment in?

awead replied:
I've moved the comment.

awead committed:
Read-only mode

When setting the appropriate environment variable, Scholarsphere will go
into "read-only" mode, during which only Solr and database reading
operations will work.  Logins are prohibited as well as any other
database writing operations.

This is a UI-level change only. Changes via the console are still
permissible.

A flash message informs the user that Scholarsphere's content cannot be
changed during this time.

srerickson reviewed (approved):
This looks good. One additional feature (part of a separate PR, probably) would be to make the banner text configurable.
